server/src/main/java/com/datastrato/gravitino/server/web/rest/RoleOperations.java

```diff
@@ -33,12 +44,76 @@
 public class RoleOperations {
   private static final Logger LOG = LoggerFactory.getLogger(RoleOperations.class);
 
+  private static final String UNSUPPORTED_ERROR_MSG = "%s is an unsupported catalog type";
+  private static final List<Privilege> CATALOG_PRIVILEGES_EXCEPT_FOR_LIST =
```

Comment:
Why don't you put the code in `core`, what is the specific reason?

Reply:
The check logic needs to get the type of catalog. We should use tree lock to get the catalog. We usually use the tree lock in the operation instead of core.

Reply:
Done.